WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingStreamsViewController.swift

```diff
@@ -0,0 +1,167 @@
+import Foundation
+
+
+/**
+*  @class           NotificationSettingStreamsViewController
+*  @brief           This class will simply render the collection of Streams available for a given
+*                   NotificationSettings collection.
+*                   A Stream represents a possible way in which notifications are communicated.
+*                   For instance: Push Notifications / WordPress.com Timeline / Email
+*/
+
+public class NotificationSettingStreamsViewController : UITableViewController
+{
+    // MARK: - View Lifecycle
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+    }
+    
+    public override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        // Manually deselect the selected row. This is required due to a bug in iOS7 / iOS8
+        tableView.deselectSelectedRowWithAnimation(true)
+    }
+    
+    
+    
+    // MARK: - Setup Helpers
+    private func setupTableView() {
+        // Empty Back Button
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: String(), style: .Plain, target: nil, action: nil)
+        
+        // Hide the separators, whenever the table is empty
+        tableView.tableFooterView = UIView()
+        
+        // Style!
+        WPStyleGuide.configureColorsForView(view, andTableView: tableView)
+    }
+    
+    
+    
+    // MARK: - Public Helpers
+    public func setupWithSettings(settings: NotificationSettings) {
+        // Title
+        switch settings.channel {
+        case let .Blog(blogId):
+            title = settings.blog?.blogName ?? settings.channel.description()
+        case .Other:
+            title = NSLocalizedString("Other Sites", comment: "Other Notifications Streams Title")
+        default:
+            // Note: WordPress.com is not expected here!
+            break
+        }
+        
+        // Structures
+        self.settings       = settings
+        self.sortedStreams  = settings.streams.sorted { $0.kind.description() > $1.kind.description() }
```

Comment:
I'll update this one, so that `self` is not required in any of the two assignments. Thank you!